target/linux/lantiq/files-4.19/arch/mips/boot/dts/lantiq/grx390_dlink_dwr-966.dts

```diff
@@ -0,0 +1,288 @@
+
+/ {
+	compatible = "dlink,dwr-966", "lantiq,xway", "lantiq,grx390";
+	model = "D-Link DWR-966";
+
+	chosen {
+		bootargs = "console=ttyLTQ0,115200";
+	};
+
+	memory@0 {
+		reg = <0x0 0x8000000>;
+	};
+
+	aliases {
+		led-boot = &power_error_red;
+		led-failsafe = &power_error_red;
+		led-running = &power_green;
+		led-upgrade = &power_error_red;
+
+		led-internet = &web_green;
+		led-wifi = &wifi_2g_green;
+	};
+
+	keys {
+		compatible = "gpio-keys-polled";
+		poll-interval = <100>;
+
+		reset {
+			label = "reset";
+			gpios = <&gpio 9 GPIO_ACTIVE_LOW>;
+			linux,code = <KEY_RESTART>;
+		};
+
+		wps {
+			label = "wps";
+			gpios = <&gpio 11 GPIO_ACTIVE_HIGH>;
+			linux,code = <KEY_WPS_BUTTON>;
+		};
+
+		ant {
+			label = "ant";
+			gpios = <&gpio 14 GPIO_ACTIVE_LOW>;
+		};
+	};
+
+	leds {
+		compatible = "gpio-leds";
+
+		wps {
+			label = "green:wps";
+			gpios = <&stp 23 GPIO_ACTIVE_LOW>;
+		};
+		web_green: web-green {
+			label = "green:internet";
+			gpios = <&stp 22 GPIO_ACTIVE_LOW>;
+		};
+		internal_ant {
+			label = "green:LTE_ANT_IND";
+			gpios = <&stp 20 GPIO_ACTIVE_LOW>;
+		};
+		3g_link {
+			label = "green:3G";
+			gpios = <&stp 15 GPIO_ACTIVE_LOW>;
+		};
+		wifi_5g {
+			label = "green:wifi_5g";
+			gpios = <&stp 14 GPIO_ACTIVE_LOW>;
+		};
+		wifi_2g_green: wifi_2g-green {
+			label = "green:wifi_2g";
+			gpios = <&stp 13 GPIO_ACTIVE_LOW>;
+		};
+		4g_link {
+			label = "green:4G";
+			gpios = <&stp 7 GPIO_ACTIVE_LOW>;
+		};
+		2g_link {
+			label = "blue:2G";
+			gpios = <&stp 4 GPIO_ACTIVE_HIGH>;
+		};
+		power_error_red: power_error-red {
+			label = "red:power_error";
+			gpios = <&stp 1 GPIO_ACTIVE_LOW>;
+		};
+		power_green: power-green {
+			label = "green:power";
+			gpios = <&stp 0 GPIO_ACTIVE_LOW>;
+		};
+		rssi_1 {
+			label = "green:rssi_5";
+			gpios = <&gpio 10 GPIO_ACTIVE_LOW>;
+		};
+		rssi_2 {
+			label = "green:rssi_4";
+			gpios = <&gpio 15 GPIO_ACTIVE_LOW>;
+		};
+		rssi_3 {
+			label = "green:rssi_3";
+			gpios = <&gpio 16 GPIO_ACTIVE_LOW>;
+		};
+		rssi_4 {
+			label = "green:rssi_2";
+			gpios = <&gpio 17 GPIO_ACTIVE_LOW>;
+		};
+		rssi_5 {
+			label = "green:rssi_1";
+			gpios = <&gpio 18 GPIO_ACTIVE_LOW>;
+		};
+	};
+};
+
+&localbus {
+	nand@1 {
+		compatible = "lantiq,nand-xway";
+		lantiq,cs = <1>;
+		bank-width = <2>;
+		reg = <1 0x0 0x2000000>;
+		#address-cells = <1>;
+		#size-cells = <1>;
+
+		partitions {
+			compatible = "fixed-partitions";
+			#address-cells = <1>;
+			#size-cells = <1>;
+
+			partition@0 {
+				label = "uboot";
+				reg = <0x00000 0x100000>;
+				read-only;
+			};
+			partition@100000 {
+				label = "ubootconfigA";
+				reg = <0x100000 0x40000>;
+				read-only;
+			};
+			partition@140000 {
+				label = "ubootconfigB";
+				reg = <0x140000 0x40000>;
+				read-only;
+			};
+			partition@180000 {
+				label = "gphyfirmware";
+				read-only;
+				reg = <0x180000 0x40000>;
+			};
+			partition@01c0000 {
+				label = "ubi";
+				reg = <0x01c0000 0x7c00000>;
```

Comment:
Please be consistent on the leading zeros for the partition location and size. Either use no leading zeros at all, or have all addresses for all partition have the same number of total digits.